### DISCUSSION TITLE
ADempiere Project Steering Meetings

### DISCUSSION THREAD

**Discussion by rvergara** [adempiere/adempiere#3918] [Steering Meetings]
#  English Version
## Meeting info
[**Original document**](https://github.com/adempiere/adempiere/files/8914090/ADempiere.Project.meeting.notes.June.10.2022.odt)
**Date:** June 10, 2022
**Time:** 19:00 GMT ~ 20:00 GMT
**Location:** Discord meeting room
**Meeting URL:** None
**Recorder:** No
**Meeting Objetive:** Review of Project and admin activities to be performed
## Participants:
- @e-Evolution 
- @marcalwestf 
- @yamelsenih 
- @eduardolopezuy 
- @rvergara 
- @piracio
- Other attendees at Discord meeting room

## Meeting agenda (Suggested by R. Vergara)
1. Discuss current project institutions
2. Define a new set of project institutions if required
3. Generate a list of all administrative activities that need to be performed
4. Generate listing of activities to support and inform the community about the project in order to achieve larger presence and collaboration

## Points discussed

1. Mario gathered information about the foundation and project directory, as per the information, none of the activities defined for both are currently being performed
3. A temporary project body has been agreed so approval of some project activities/tasks can be achieved. This body, for the time being will comprise of current major contributors, namely Victor Perez, Yamel Senih and Mario Calderon.
R. Vergara Volunteered to act as Per Tempore Secretary for, at least, until the next meeting. This was approved by the group of major contributors
R Vergara suggested that decisiones of the majors contributors groups should be taken by unanimity to avoid vetoes or exclusions. This point is still being considered by the major contributors group.
4. A list of administrative activities include:
    - Define where to establish the project foundation (Germany, Canada, others)
This was discussed and agreed to review in a future date when a major contributor activity is achieved
    - Decide on how to manage project assets (Branding, domains, etc) 
This was agreed to be discussed in following project meetings
    - Define priority activities that could be funded by current funds
This was agreed to be discussed in future meetings
    - Review correct operation of current sites (wiki, project web page, demo site, twitter, others) 
The major contributors groups asked for volunteers to contribute with these activities
5. A list of activities to support and inform the community includes:
    - Reactivate the wiki or find a better tool to maintain the project documentation. 
Call for volunteers to execute this task was requested
    - Define and organize a tool and site to locate the project knowledge base
It was suggested that a volunteer should review the options that Github may provide to resolve this task.
    - Organize and execute tutorials and/or seminar to provide ADempiere information to the community
A group of volunteers is required to execute this task
    - Organize an information campaign for the upcoming release of ADempiere 3.9.4
R. Vergara and E. Lopez volunteered to execute this activity

## Next meeting
Next follow up meeting has been scheduled on June 24, 2022 at 19:00 GMT via Discord

#  Versión Español
## Información de Reunión
[**Documento Original**](https://github.com/adempiere/adempiere/files/8914090/ADempiere.Project.meeting.notes.June.10.2022.odt)
**Fecha:** 10 de Junio 2022
**Hora:** 19:00 GMT ~ 20:00 GMT
**Lugar:** Discord
**URL de la Reunión:** None
**Grabación:** No
**Objetivo de la Reunión:** Revisión del Proyecto y actividades administrativas por realizar
## Participantes:
- @e-Evolution 
- @marcalwestf 
- @yamelsenih 
- @eduardolopezuy 
- @rvergara 
- @piracio
- Otros asistentes en la sala de reuniones de Discord

## Agenda de Reunión (Sugerida por R. Vergara)
1. Discutir las instituciones del proyecto actual
2. Definir  un nuevo conjunto de instituciones del proyecto, si es requerido
3. Generar una lista de todas las actividades administrativas que necesitan ser realizadas
4. Generar una lista de actividades para apoyar e informar a la comunidad acerca del proyecto para lograr una mayor presencia y colaboración

## Puntos discutidos

1. Mario recopiló información sobre la fundación y el directorio del proyecto, según la información, actualmente no se está realizando ninguna de las actividades definidas para ambos.
3. Se acordó un organismo de proyecto temporal para que se pueda lograr la aprobación de algunas actividades / tareas del proyecto. Este organismo, por el momento, estará compuesto por los principales contribuyentes actuales, a saber, **Víctor Pérez**, **Yamel Senih** y **Mario Calderón**.
R. Vergara se ofreció como voluntario para actuar como secretario interino, por lo menos, hasta la próxima reunión. Esto fue aprobado por el grupo de principales contribuyentes
R Vergara sugirió que las decisiones de los grupos de mayores contribuyentes se tomen por unanimidad para evitar vetos o exclusiones. Este punto aún está siendo considerado por el grupo de principales contribuyentes.
4. La lista de las actividades administrativas incluye lo siguiente:
    - Definir dónde se establecerá la fundación del proyecto (Alemania, Canada, otros)
Esto fue discutido y se acordó revisar en una fecha futura cuando se logre una actividad de contribuyentes importantes.
    - Decidir cómo manejar los activos del proyecto (Marca, dominios, entre otros) 
Se acordó que será discutido en la siguientes reuniones del proyecto
    - Definir actividades de prioridad que podrían financiarse con los fondos actuales
Se acordó discutir este punto en las próximas reuniones
    - Verificar la operación correcta de los sitios actuales (wiki, página web del proyecto, sitio de demostración, twitter, otros) 
Los principales grupos de contribuyentes pidieron voluntarios para contribuir con estas actividades.
5. La lista de actividades a apoyar e informar a la comunidad incluye lo siguiente:
    - Reactivar el wiki o buscar una mejor herramienta para mantener la documentación del proyecto. 
Se solicitó convocatoria de voluntarios para ejecutar esta tarea
    - Definir y organizar una herramienta y sitio para colocar la base de datos de conocimiento del proyecto
Se sugirió que un voluntario revisara las opciones que Github puede proporcionar para resolver esta tarea.
    - Organizar y ejecutar tutoriales y / o seminarios para proveer indormación sobre ADempiere a la comunidad
Un grupo de voluntarios es requerido para ejecutar esta tarea
    - Organizar una campaña de información para el próximo lanzamiento de ADempiere 3.9.4
R. Vergara y E. Lopez serán los voluntarios para esta tarea

## Próxima reunión
La siguiente reunión se agendó para el 24 de Junio del 2022 a las 19:00 GMT via Discord

**Comment by yamelsenih** [comment]
- [ ] Reactivar el wiki o buscar una mejor herramienta para mantener la documentación del proyecto.
Se solicitó convocatoria de voluntarios para ejecutar esta tarea

**Reply by yamelsenih** [reply]
Propongo usar la documentación que se genera desde ADempiere y que es muy potente, la misma ya construye sola [aquí](https://adempiere.github.io/docs/). Para usarla propongo lo siguiente:
- Hacer una reunión básica de inicio para explicar lo que se puede hacer con la plataforma
- Definir los encargados o responsables de cargar contenido
- Definir tiempos de entrega y reuniones de seguimiento
- Definir un procedimiento para aprobación y prueba de la documentación

Como referencia [reStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) es ampliamente usado por ser multi-idioma y muy potente y algunos sitios de referencia se pueden ver aquí:

- [Open edX](https://edx.readthedocs.io/projects/edx-developer-docs/en/latest/micro_frontends/index.html)
- [Apache CouchDB](https://docs.couchdb.org/en/stable/)

**Reply by e-Evolution** [reply]
Hola Yamel una pregunta la documentación basada en reStructuredText se puede desplegar como un sitio estático en https://pages.github.com/ ? 

Si esto es posible excelente una preocupación menos.

La razón de usar Github Pages es que tenemos el servicio ya incluido en el proyecto ADempiere y aquí ahorrar costos es muy importante. https://youtu.be/2MsN8gpT6jY

Por que es importante usar   Github Pages.
- No tiene costo adicional el hosting del contenido ya que es un servicio de github.
- Generalmente la documentación esta basada en un estándar como Markdown , así aprendiendo una simple sintaxis se pueden obtener resultados loables.
- Se mencionado en múltiples oportunidades que los contribuidores que no son desarrolladores no tienen como contribuir por que no escriben código, pero al usar Github y Github Pages el ciclo de contribución en la documentación sería muy similar al del código, es decir alguien hace un commit de alguna pagina documentando una funcionalidad , otro contribuidor hace la revisión por pares y finalmente se hace un merge de su contenido, esto permitirá alimentar la estadística de Github y tener una manera cuantitativa de medir las contribuciones de los documentadores 
- Actualmente la mayoría de los que participamos  sabemos utilizar ya las herramientas de Github.
- No se requiere demasiado tiempo para administrar estos recursos o experiencia.

Lo que más me interesa a mi es por dar crédito y reconocer la contribución de manera cuantitativa y usar Github y Github Page son perfectos para este propósito.

Si la propuesta actual no es posible usar con Github Pages deberíamos buscar algo que si sea soportado como :

https://vuepress.vuejs.org/
https://docsify.js.org/#/
https://jekyllrb.com/

Posdata : Entre menos personas se requiera para administrar esos servicios mejor.

**Reply by yamelsenih** [reply]
Hola Víctor, claro mira aquí ya lo despliegue, solo falta que las personas se comprometan a cargar contenido. Estoy a la orden y es gratis.

https://adempiere.github.io/docs/


De ti solo necesito un subdominio docs.adempiere.io y docs.adempire.net apuntando a cualquiera de estas IP's para que usemos esto como la documentación de ADempiere y sin el dominio de github.

Te dejo las IP's
```
185.199.108.153
185.199.109.153
185.199.110.153
185.199.111.153
```

**Reply by yamelsenih** [reply]
Avísame para cambiar el CName y que use ese subdominio. Nota: es gratis y usa las.paginas estáticas de github

**Reply by e-Evolution** [reply]
@yamelsenih excelente, Una pregunta esas ips de donde son o en que servidor esta desplegado eso , por que la razón de ir con Github Page es que Github se encarga de la infraestructura sin necesidad de ninguna intervención , sys admin o cualquier otra cosa, no tiene caso estar pagando por un servidor que en el momento que falle hay que estar buscando al responsable o a saber si se pago la factura , todas esas cosas son las que hay que evitar , dado que es un problema recurrente con el actual servidor wiki y website.

Github Page permite montar sitios estáticos y con Github Action se automatiza la publicación , lo cual simplifica el trabajo y la administración.

Por favor puedes integrarlo siguiendo estos pasos :

https://pages.github.com/

Por otra parte tal vez sea bueno un pequeño video que explique el flujo de como construir y publicar cualquier documentación para que a la brevedad los voluntarios puedan  contribuir fácilmente.

Nos puedes ayudar con eso, gracias ?

**Reply by yamelsenih** [reply]
Ya esta integrado así jejeje. Esas IPS son de github para la carga con dominio propio

**Reply by yamelsenih** [reply]
Aquí está la referencia https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain

**Comment by yamelsenih** [comment]
- [ ] Definir y organizar una herramienta y sitio para colocar la base de datos de conocimiento del proyecto
Se sugirió que un voluntario revisara las opciones que Github puede proporcionar para resolver esta tarea.

**Reply by yamelsenih** [reply]
Para esto mi propuesta es usar https://stackoverflow.com/ como una base de datos de conocimiento

**Comment by yamelsenih** [comment]
- [ ] Verificar la operación correcta de los sitios actuales (wiki, página web del proyecto, sitio de demostración, twitter, otros) 
Los principales grupos de contribuyentes pidieron voluntarios para contribuir con estas actividades.

**Reply by yamelsenih** [reply]
A pesar de contar con recursos propuestos por @piracio para ADempiere en cuanto a infraestructura, desconozco si se pueden usar para este fin. Creo que deberíamos tener un sitio facil de construir y que no requiera de un técnico especializado para liberar versiones.
Mi propuesta es muy sencilla:
- Crear una infraestructura básica basada en [docker](https://www.docker.com/)
- Gestionar los despliegues con [Argo CD](https://argo-cd.readthedocs.io/en/stable/): Esto permitirá hacer los despliegues de manera automática
- Montar un sitio web con [WordPress](https://wordpress.com/): para esto necesito:
  - Permiso de @piracio para acceder y montarlo en los servidores
  - Una reunión con @eduardolopezuy y @rvergara para copiar el contenido de la página actual
  - Una instancia con la página actual al menos por una semana para copiar el contenido completamente
  - Un tiempo de una semana para tener un contenido base montado 
- Definir los encargados o responsables de cargar contenido

Para esta propuesta propongo a @carlosaparadam

**Reply by e-Evolution** [reply]
> Montar un sitio web con [WordPress](https://wordpress.com/): para esto necesito:

Creo que quede bastante decepcionado de [WordPress](https://wordpress.com/):  , incluso de Liferay , requieren mantener una infraestructura y no tienen una manera simple de manejar el control de cambios.

Por los mismos desventajas y ventajas de la documentación , usemos Github Pages , aquí hay un ejemplo de un sitio creado estáticamente https://www.sketch.com/ y no requiere un servidor , base de datos , simplemente se crea el proyecto en Github Page y se sigue el flujo de publicación.

Problemas con los que nos encontramos actualmente:

1.- Se requiere la renta de un servidor y quien lo administre
2.- Se requiere instalar wordpress 
3.- Se requiere montar una base de datos.
4.- Se requiere mantener actualizado , instalar regularmente parches 
5.- Se requiere desplegar una seguridad perimetral para que no hacken el server.

Basado en mi historia , el servidor de base de datos se cae regularmente, vi por experiencia quien lo uso y prácticamente destruyeron su sitio web y hackearon el wordpress, lo que refleja que este es de los CMS más inseguros , lo mismo que Liferary , Se que hay manera de asegurarlos y mantenerlos seguros pero requieres algún experto de seguridad que haga ese trabajo.

Adempiere requiere algo simple un sitio estático que sea fácil de mantener  con simples commits y ademas si usamos github como repositorio se puede manejar un control de cambios y reconocer de manera cuantitativa la contribución de los colaboradores.

Así requerimos un colaborador que cree un website basado en Github Page , ya que no significa un costo adicional para el proyecto.

**Comment by piracio** [comment]
Hola @yamelsenih , no apezco como participalte en la reunion.

**Reply by e-Evolution** [reply]
> Hola @yamelsenih , no apezco como participalte en la reunion.

@piracio , ya te agregue en la lista de participantes

**Reply by yamelsenih** [reply]
Hola @piracio no te agregué para no ser incongruente con lo que escribió Ramiro disculpa.
![Screenshot from 2022-06-24 11-32-25](https://user-images.githubusercontent.com/2333092/175568660-6a9ce8be-9e61-4095-9547-7ed4a2cd4148.png)
